ledger/participant-integration-api/src/test/suite/scala/platform/InMemoryStateSpec.scala

```diff
@@ -46,64 +48,24 @@ class InMemoryStateSpec extends AsyncFlatSpec with MockitoSugar with Matchers {
       val initLedgerEnd = ParameterStorageBackend
         .LedgerEnd(initOffset, initEventSequentialId, initStringInterningId)
 
-      val updateStringInterningView = mock[(UpdatingStringInterningView, LedgerEnd) => Future[Unit]]
       when(updateStringInterningView(stringInterningView, initLedgerEnd)).thenReturn(Future.unit)
+      when(dispatcherState.stopDispatcher()).thenReturn(Future.unit)
 
-      when(dispatcherState.reset(initLedgerEnd)).thenReturn(Future.unit)
-      when(dispatcherState.initialized).thenReturn(true)
       for {
         // INITIALIZED THE STATE
         _ <- inMemoryState.initializeTo(initLedgerEnd)(updateStringInterningView)
-
+      } yield {
         // ASSERT STATE INITIALIZED
-        _ = {
-          inMemoryState.initialized shouldBe true
-
-          verify(contractStateCaches).reset(initOffset)
-          verify(transactionsBuffer).flush()
-          verify(mutableLedgerEndCache).set(initOffset, initEventSequentialId)
-          verify(updateStringInterningView)(stringInterningView, initLedgerEnd)
-          verify(dispatcherState).reset(initLedgerEnd)
-        }
-
-        reInitOffset = Offset.fromHexString(Ref.HexString.assertFromString("abeeee"))
-        reInitEventSequentialId = 9999L
-        reInitStringInterningId = 50
-        reInitLedgerEnd = ParameterStorageBackend
-          .LedgerEnd(reInitOffset, reInitEventSequentialId, reInitStringInterningId)
-
-        // RESET MOCKS
-        _ = {
-          reset(
-            mutableLedgerEndCache,
-            contractStateCaches,
-            transactionsBuffer,
-            updateStringInterningView,
-          )
-          when(updateStringInterningView(stringInterningView, reInitLedgerEnd)).thenReturn(
-            Future.unit
-          )
-          when(dispatcherState.reset(reInitLedgerEnd)).thenReturn(Future.unit)
-        }
 
-        // RE-INITIALIZE THE STATE
-        _ <- inMemoryState.initializeTo(reInitLedgerEnd) {
```

Comment:
This test is good but shouldn't we additionally/still have a (re) `initiaizeTo` test.  The difference with this PR and the wider ranging PR is that in this scenario all in-memory state is lost on re-initialization.

Reply:
Left it out because it's addressed in the PR where this was split from. Added it back.